rust-sdk/core/src/math/tick_array.rs

```diff
@@ -0,0 +1,196 @@
+use crate::{TickFacade, TickArrayFacade, TICK_ARRAY_SIZE};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct TickArraySequence {
+    tick_arrays: [TickArrayFacade; 3],
+    tick_spacing: u16,
+}
+
+impl TickArraySequence {
+    pub fn new(one: TickArrayFacade, two: TickArrayFacade, three: TickArrayFacade, tick_spacing: u16) -> Self {
```

Comment:
swap v2 takes in additional tick-arrays via the remaining_accounts. So potentially we can support n-many tick-arrays here (limited by compute units / tx-size). I believe the tick-arrays can also be uninitialized with sparse-swap. 

1st iteration is fine but a heads up we may want to take in a Vec for the current swap-quote & may want to have a variant of swap-quote that also deals with finding the right tick-array sequence to plug in here

Reply:
From what I understand in the new swap instruction 3 TA can be specified directly with the option to add up to 3 more through `remaining_accounts`. The swap instruction itself however will only ever use 3 TAs at max